src/mavis/convert/vcf.py

```diff
@@ -122,6 +123,85 @@ def parse_bnd_alt(alt: str) -> Tuple[str, int, str, str, str, str]:
         raise NotImplementedError('alt specification in unexpected format', alt)
 
 
+def convert_imprecise_break(std_row: Dict, record: List[VcfRecordType], bp_end: int) -> Dict:
+    """
+    Handles IMPRECISE calls, that leveraged uncertainty from the CIPOS/CIEND/CILEN fields.
+
+    bp1_s = breakpoint1 start
+    bp1_e = breakpoint1 end
+    bp2_s = breakpoint2 start
+    bp2_e = breakpoint2 end
+
+    Insertion and deletion edge case - in which bp1_e > bp2_s
+    E.g bp1_s = 1890, bp1_e = 2000, bp2_s = 1900, bp2_e = 1900.
+    break1 ------------------------=======================--------------
+    break2 ------------------------==========---------------------------
+
+    Insertion edge case - in which bp1_e > bp1_s
+    E.g bp1_s = 1890, bp1_e = 1800, bp2_s = 1800, bp2_e = 1800.
+    break1 ------------------------==-----------------------------------
+    break2 ------------------------=------------------------------------
+
+    Insertion edge case - in which bp1_s > bp2_s
+    E.g bp1_s = 1950, bp1_e = 2000, bp2_s = 1900, bp2_e = 3000.
+    break1 ------------------------==-----------------------------------
+    break2 -----------------------========------------------------------
+    """
+
+    try:
+        if record.info.get(
+            'CILEN'
+        ):  # as per https://github.com/samtools/hts-specs/issues/615, CILEN takes priority over CIPOS in dictating breakpoint2
+            std_row.update(
+                {
+                    COLUMNS.break1_position_start: max(
+                        1, record.pos + record.info.get('CIPOS', (0, 0))[0]
+                    ),
+                    COLUMNS.break1_position_end: record.pos + record.info.get('CIPOS', (0, 0))[1],
+                    COLUMNS.break2_position_start: max(
+                        1, bp_end + record.info.get('CILEN', (0, 0))[0]
+                    ),
+                    COLUMNS.break2_position_end: bp_end + record.info.get('CILEN', (0, 0))[1],
+                }
+            )
+        else:
+            std_row.update(
+                {
+                    COLUMNS.break1_position_start: max(
+                        1, record.pos + record.info.get('CIPOS', (0, 0))[0]
+                    ),
+                    COLUMNS.break1_position_end: record.pos + record.info.get('CIPOS', (0, 0))[1],
+                    COLUMNS.break2_position_start: max(
+                        1, bp_end + record.info.get('CIEND', (0, 0))[0]
+                    ),
+                    COLUMNS.break2_position_end: bp_end + record.info.get('CIEND', (0, 0))[1],
+                }
+            )
+
+        if (
```

Comment:
I think you might be able to simplify this part a bit

Is there a reason we are only doing this for insertions? it probably makes sense for all event types so long as they are intrachromosomal? in which case you could remove the sv  type check entirely and just check the chromosomes match

then use min/max to get the positions

```python
if std_row["break1_chromosome"] == std_row["break2_chromosome"] and Interval.overlaps(
    (std_row['break1_position_start'], std_row['break1_position_end']), 
    (std_row['break2_position_start'], std_row['break2_position_end'])
):

  std_row['break2_position_start'] = max(std_row['break1_position_start'], std_row['break2_position_start'])
  std_row['break1_position_end'] = min(std_row['break1_position_end'], std_row['break2_position_end'])
```

this basically just says break1 end should get cut off at the end of break2 if it goes past it and break2 start should get cut off at the start of break2 if it starts before it

Reply:
If we don't include SV type information, this call [Manta BND](https://github.com/bcgsc/mavis/blob/master/tests/test_mavis/convert/test_tool.py#L305 ) will always be "fixed". We can just avoid BND calls for now as a workaround.